teamcity/messages.py

```diff
@@ -10,6 +10,14 @@
     text_type = str
 
 
+quote = {"'": "|'", "|": "||", "\n": "|n", "\r": "|r", ']': '|]'}
+
+def escapeValue(value):
+    return "".join([quote.get(x, x) for x in value])
```

Comment:
This code was existing code that was moved. I went ahead and used the generator expression in 3056ff4. I figure that @shalupov can cherry-pick if he wants some commits but not others.